### ISSUE TITLE
Support optimizing/saving trainable parameters in the loss module

### ISSUE THREAD

**Issue by ntubiolin** [amjltc295/pytorch-golden-template#76]
Recently I am trying to use one optimizable loss function from "A General and Adaptive Robust Loss Function" Jonathan T. Barron, CVPR 2019
However, the current template seems not support the optimization of loss module. Will this template support this feature?
